sc/social/like/plugins/facebook/templates/metadata.pt

```diff
@@ -3,7 +3,7 @@
 </tal:comment>
 <tal:fb condition="not:view/is_plone_5">
   <meta property="og:site_name" tal:attributes="content view/portal_title" />
-  <meta property="og:url" tal:attributes="content view/url" />
+  <meta tal:condition="view/canonical_url" property="og:url" tal:attributes="content view/canonical_url" />
```

Comment:
Also this would break the current behavior and hide og:url meta tag.. how about to add a new line with:
```html
<meta tal:condition="not:view/canonical_url" property="og:url" tal:attributes="content view/url" />
```